templates/mdd/task3_qarecs/templates.yaml

```diff
@@ -0,0 +1,84 @@
+dataset: mdd
+subset: task3_qarecs
+templates:
+  1614890b-362c-4ee8-850d-841cf511d169: !Template
+    id: 1614890b-362c-4ee8-850d-841cf511d169
+    jinja: "Speaker 1: {{dialogue_turns.utterance[0]}}
+
+
+      Speaker 2: {{dialogue_turns.utterance[1]}}
+
+
+      Speaker 1: {{dialogue_turns.utterance[2]}}
+
+
+      Speaker 2:|||{{dialogue_turns.utterance[3]}}"
+    name: next_utterance_4
+    reference: Given the first three dialogues,  generate next utterance.
+  3e5a19e5-aa33-467a-bcd2-f84d99f32759: !Template
+    id: 3e5a19e5-aa33-467a-bcd2-f84d99f32759
+    jinja: "{{dialogue_turns.utterance[0]}}|||{{dialogue_turns.utterance[1]}}"
+    name: recommend_movie_1
+    reference: Given likes, recommend a movie.
+  76888d6e-76fa-47db-a8b3-9980f082df51: !Template
+    id: 76888d6e-76fa-47db-a8b3-9980f082df51
+    jinja:
+      "{% if dialogue_turns.utterance|length==6 %}{{dialogue_turns.utterance[4]}}|||{{dialogue_turns.utterance[5]}}{%
+      else %}|||{% endif %}"
+    name: recommend_movie_2
+    reference:
+      Given a single preference, recommend a movie. Only works for dialogues
+      with 6 utterances.
+  91f33bcf-3c0e-49e6-ae86-28f77e224734: !Template
+    id: 91f33bcf-3c0e-49e6-ae86-28f77e224734
+    jinja:
+      "{% if dialogue_turns.utterance|length==6 %} {{dialogue_turns.utterance[1]}}
+
+
+      {{dialogue_turns.utterance[3]}}
+
+
+      {{dialogue_turns.utterance[4]}}|||{{dialogue_turns.utterance[5]}}{% else %}|||{%
+      endif %}"
+    name: recommend_movie_3
+    reference:
+      Given previous suggestion, some movie description, and preference,
+      recommend a movie.
+  92cb9273-b89f-410f-a7f2-db8c84f42862: !Template
+    id: 92cb9273-b89f-410f-a7f2-db8c84f42862
+    jinja:
+      "{% if dialogue_turns.utterance|length==6 %}{{dialogue_turns.utterance[0]}}
+
+
+      {{dialogue_turns.utterance[4]}}|||{{dialogue_turns.utterance[5]}}{% else %}|||{%
+      endif %}"
+    name: recommend_movie_4
+    reference: Given likes and preferences, recommend a movie.
```

Comment:
I find partial history templates kind of weird. Not sure I would be able to answer it correctly myself even if I knew a lot about films (`recommend_movie_3` and `recommend_movie_4`). I think there is a lot of interesting reasons to encourage the model to process long inputs/conditionings. I would remove these (3 and 4)

Reply:
The data has this pattern:
- Speaker 1 describes likes, asks for a movie.
- Speaker 2 recommends a movie.
- Speaker 1 asks a question about the movie.
- Speaker 2 answers the question.
[Optionally]
- Speaker 1 tells preferences and asks for another recommendation.
- Speaker 2 recommends another movie.

Why I thought 3 and 4 are interesting:
- In 3, the movie name and its description are provided, then a preference is provided over this recommended movie, and then a movie is recommended.
- In 4, the likes, preferences are provided, and then a movie is to be recommended.

I agree with you on the point of being able to answer it correctly. But, if a model is able to predict the next utterance using the entire history, should it also not be able to do it with these details to a certain extent?

Reply:
Right, i think the interesting part of dialogue is generating the next utterance in reaction to the grounding in a coherent and logical way, looking at the examples from `recommend_movie_3` and `recommend_movie_4`, it's hard to find the consistency:

`recommend_movie_3`
<img width="553" alt="Capture d’écran 2021-06-10 à 11 23 48 PM" src="https://user-images.githubusercontent.com/16107619/121626161-eabf7400-ca42-11eb-9e52-038b5441eed4.png">
these look like random concatenation of random entities.

`recommend_movie_4`
<img width="612" alt="Capture d’écran 2021-06-10 à 11 21 16 PM" src="https://user-images.githubusercontent.com/16107619/121625996-90261800-ca42-11eb-8be7-4a1863027909.png">
The history is destroyed and it looks very like noise (why more? why other suggestions?)

Reply:
you need to give more details about what these specific utterances expresse and how the next utterance is connected to these (the reasoning part). for instance this could be
`Given this celebrities name {{uterance[0]}}, and the fact that I like more {{utterance[3]}}, provide a recommendation |||`

Reply:
Since the each utterance is complete in itself, can I do something like:
`recommed_movie_3`
```
I am aware that the movie {{utterance[1]}} is about {{utterance[3]}}. However, {{utterance[4]}}
|||
{{utterance[5]}}
```
and 
`recommend_movie_4`
```
{{utterance[0]}} Also, {{utterance[4]}}
|||
{{utterance[5]}}
```

Reply:
Another way to explicitly state that a movie is to be recommended:
`
recommed_movie_3
`
```
{{ ["My friend is ", "I am", "My colleague is", "A person is "]|random }} aware that the movie {{utterance[1]}} is about {{utterance[3]}}. However, <if "I am" is not selected> they said: <endif>{{utterance[4]}}. Can you recommend a movie for <if "I am" is not selected> them <else> me <endif>?
|||
{{utterance[5]}}
```
and
`recommend_movie_4`

```
{{['','A friend said:', 'A colleague said:'| 'A person said:'] | random}}{{utterance[0]}} Also, <choose appropriate phrase here>: {{utterance[4]}}. Recommend a movie for <if "" is not selected> them <else> me <endif>.
|| 
{{utterance[5]}}
```
First one is empty because it's a first-person statement in `recommend_movie_4`.

Reply:
let's have both, i like them both